### PR TITLE
add *.xtext.ide project and FormattingService1

### PR DESCRIPTION
**Pull request by hasu**
That is, add an LSP Xtext IDE project.  In it include a simple
FormattingService subclass that supports INodeModelFormatter instead of
IFormatter2 for compatibility with existing IFormatter and ITokenStream
and Format DSL based formatter implementations. In that class use code
from an earlier version of FormattingService.

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.xtext.ide
+Bundle-SymbolicName: com.avaloq.tools.ddk.xtext.ide;singleton:=true
+Bundle-Version: 8.3.0.qualifier
+Bundle-Vendor: Avaloq Evolution AG
+Require-Bundle: org.eclipse.xtext.ui,
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF:7]: Do we really need a dependency on org.eclipse.xtext.ui? I though the IDE plug-ins are independent on the UI plugins, and if anything the UI plugins are the ones that depend on the IDE ones.

File: com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ org.eclipse.lsp4j,
+ com.avaloq.tools.ddk.xtext
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF:9]: Why is  com.avaloq.tools.ddk.xtext needed? I do not see any dependency in the code for this one.

File: com.avaloq.tools.ddk.xtext.ide/.project [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		<buildCommand>
+			<name>edu.umd.cs.findbugs.plugin.eclipse.findbugsBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.xtext.ui.shared.xtextNature</nature>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.ide/.project:47]: I think we need to remove the org.eclipse.xtext.ui.shared.xtextNature nature

File: com.avaloq.tools.ddk.xtext.ide/.project [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.xtext.ui.shared.xtextBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.ide/.project:27]: I think we need to remove the xtext builder

File: com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: com.avaloq.tools.ddk.xtext.ide
+Bundle-SymbolicName: com.avaloq.tools.ddk.xtext.ide;singleton:=true
+Bundle-Version: 8.3.0.qualifier
+Bundle-Vendor: Avaloq Evolution AG
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF:6]: Other manifest have the line
Bundle-ActivationPolicy: lazy
shall we not have it here as well?

File: com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF (hunk 1 above)
Comment (hasu) [reply com.avaloq.tools.ddk.xtext.ide/META-INF/MANIFEST.MF:7]: Right. Should be org.eclipse.xtext.ide, probably.

**Review by rubenporras — APPROVED** [review]

**Pull request merged**